app/furniture/marketplace/delivery_areas/index.html.erb

```diff
@@ -24,11 +24,7 @@
       <%- end %>
     </div>
     <div class="text-center mt-3">
-      <%- if params[:archive] %>
-        <%= link_to "Active Delivery Areas", marketplace.location(child: :delivery_areas) %>
-      <%- else %>
-        <%= link_to "Archived Delivery Areas", marketplace.location(child: :delivery_areas, query_params: { archive: true }) %>
-      <%- end %>
+      <%= render Marketplace::Archivable::IndexLinkComponent.new(marketplace: marketplace, resource: :delivery_areas, to_archive: !params[:archive]) %>
```

Comment:
Will more logic go into this component? It seems far too granular to me because this component is essentially a one-liner if you cast the param to a bool and just pass it in, along with the new translations:

```ruby
<%= link_to 
	t("delivery_areas.#{!params[:archive] ? :archive : :active}"), 
	marketplace.location(
		child: :delivery_areas, 
		query_params: { 
			archive: !params[:archive] 
		}
	) 
%>
```
Unless rails will do something weird with the empty value in the `query_parms` dictionary? I think your gut is right that the abstraction is at a higher level.

Reply:
Yea, that's part of why I think a Smerter :tm:  Abstraction would be a `Marketplace::Arhcivable::ManagementComponent` or something.

Initially I componentized it to get some tests around it; since I wanted to make sure I was looking up the right translation in the right context.